contracts/connection-router/src/contract/query.rs

```diff
@@ -1,17 +1,41 @@
+use cosmwasm_std::{Deps, Order};
+use cw_storage_plus::Bound;
+
 use connection_router_api::error::Error;
 use connection_router_api::{ChainEndpoint, ChainName};
-use cosmwasm_std::Deps;
 use error_stack::{Result, ResultExt};
 
 use crate::state::chain_endpoints;
 
+// Pagination limits
+const MAX_LIMIT: u32 = 50;
+const DEFAULT_LIMIT: u32 = 10;
```

Comment:
Why not default to max u32?

Reply:
@cgorenflo If the calling contract isn't aware of that pagination and the default is 10, it will not get all the results which may lead to faulty behaviour. If the default is max u32, the calling contract would run out of gas. Either way requires a re-deployment of the calling contract to fix the issue so I am not sure what is it to be avoided here. Former sounds worse to me.

Reply:
fair point, lets go with max u32 then

Reply:
in that case I think both should be set to max u32

Reply:
removed the max as default to max u32, can add back if needed in the future